### PR TITLE
[9.0][MIG] quality_control

### PR DESCRIPTION
**Pull request by LoisRForgeFlow**
Continue the works started in https://github.com/OCA/manufacture/pull/100.

### CODE REVIEW

**Review by MiquelRForgeFlow — APPROVED** [review]
LGTM :+1: 
But I think some of the previous commits could be squashed.

**Comment by lreficent** [comment]
Cherry-picked fix done in https://github.com/OCA/manufacture/pull/205/commits

**Comment by LoisRForgeFlow** [comment]
Cherry-picked fix done in https://github.com/OCA/manufacture/pull/205

File: quality_control/wizard/qc_test_wizard.py
Comment (andhit-r) [inline quality_control/wizard/qc_test_wizard.py]: Change into short header

File: quality_control/wizard/__init__.py
Comment (andhit-r) [inline quality_control/wizard/__init__.py]: Change into short header

File: quality_control/models/qc_trigger_line.py
Comment (andhit-r) [inline quality_control/models/qc_trigger_line.py]: Change into short header

File: quality_control/models/qc_trigger.py
Comment (andhit-r) [inline quality_control/models/qc_trigger.py]: Change into short header

File: quality_control/models/qc_test_category.py
Comment (andhit-r) [inline quality_control/models/qc_test_category.py]: Change into short header

File: quality_control/models/qc_test.py
Comment (andhit-r) [inline quality_control/models/qc_test.py]: Change into short header

File: quality_control/models/qc_inspection.py
Comment (andhit-r) [inline quality_control/models/qc_inspection.py]: Change into short header

File: quality_control/models/product_template.py
Comment (andhit-r) [inline quality_control/models/product_template.py]: Change into short header

File: quality_control/models/product_product.py
Comment (andhit-r) [inline quality_control/models/product_product.py]: Change into short header

File: quality_control/models/product_category.py
Comment (andhit-r) [inline quality_control/models/product_category.py]: Change into short header

File: quality_control/models/__init__.py
Comment (andhit-r) [inline quality_control/models/__init__.py]: Change into short header

File: quality_control/__init__.py
Comment (andhit-r) [inline quality_control/__init__.py]: Change into short header

**Review by andhit-r — CHANGES_REQUESTED** [review]
Code review

**Comment by LoisRForgeFlow** [comment]
Open PRs in v8:
* [x] : https://github.com/OCA/manufacture/pull/193
* [x] : https://github.com/OCA/manufacture/pull/213 (merged and added [here](https://github.com/OCA/manufacture/pull/198/commits/784cce1fed74758f8042f56c1d181dc53baa51cf))

**Comment by LoisRForgeFlow** [comment]
@andhit-r Is this ok for you now? could you update your review?

**Comment by LoisRForgeFlow** [comment]
@andhit-r ping

**Comment by LoisRForgeFlow** [comment]
@OCA/manufacturing-maintainers can we get an update on this?

**Comment by damendieta** [comment]
Hi, on creation of the Test Categories there there is an error that tries to compute the name but as It's a new record, the value is still false:

File "/mnt/extra-addons/manufacture/quality_control/models/qc_test_category.py", line 23, in _get_complete_name
    self.complete_name = " / ".join(reversed(names))
TypeError: sequence item 0: expected string, bool found

Replacing:
names = [self.name]
With:
names = [self.name or '']

Solves the issue.

Daniel.

**Comment by damendieta** [comment]
Hi, I already fixed it and added some minor changes that I needed to test the module.

I made a PR with the changes:
https://github.com/Eficent/manufacture/pull/2

Also, could you please explain the logic of triggers on the readme, I coudn't find a way to make triggers work.

Thanks a lot.

**Comment by LoisRForgeFlow** [comment]
Hi @damendieta 

First of all: thanks for your review!

Regarding the bug you reported, it is already being fixed in v8 and waiting to be ported to this PR after its merge, as noted here: https://github.com/OCA/manufacture/pull/198#issuecomment-312675900. Please review that PR (https://github.com/OCA/manufacture/pull/193) in order to move forward.

About the qc triggers, let me clarify that to make use of them you need either [quality_control_mrp](https://github.com/OCA/manufacture/tree/8.0/quality_control_mrp) or [quality_control_stock](https://github.com/OCA/manufacture/tree/8.0/quality_control_stock) modules. However, they still need to be migrated to v9.

Your help would be appreciated :wink:

**Comment by damendieta** [comment]
@lreficent ok, I approved the PR #193, what else do we need for it to get merged?

**Comment by LoisRForgeFlow** [comment]
@damendieta fix from https://github.com/OCA/manufacture/pull/193 added. Now you can review this PR and approve it if everything is ok for you.

**Comment by eLBati** [comment]
@lreficent  hi, could you check commits like https://github.com/OCA/manufacture/pull/198/commits/e6d89619eeab5355d9c60887851c055b4a048acb with no message and unrecognised author?
Also please check if some commits can be squashed

**Comment by LoisRForgeFlow** [comment]
Hi @eLBati  

It looks like "alfredo" has changed either his github email or name and that's why the author is unrecognized, however I don't know his current github account (If someone could point it to me I'll change the commit author accordingly). Regarding the squashing, I've already squashed consecutive transifex commits, I don't know if I can squash further since I'm not the author of any of the previous versions commits.

**Comment by eLBati** [comment]
Maybe @oihane or @pedrobaeza know about his account?

**Comment by oihane** [comment]
Alfredo de la Fuente's github user is alfredoavanzosc

**Comment by LoisRForgeFlow** [comment]
@oihane Thanks!

@eLBati history has been fixed.

File: quality_control/views/product_category_view.xml
Comment (eLBati) [inline quality_control/views/product_category_view.xml]: Please use odoo

**Review by eLBati — APPROVED** [review]
Just a remark

**Review by SimoRubi — COMMENTED** [review]
Just code review, looks good

File: quality_control/views/product_category_view.xml
Comment (LoisRForgeFlow) [reply quality_control/views/product_category_view.xml]: Done, Thanks!

**Comment by LoisRForgeFlow** [comment]
@SimoRubi Hi, please use the option to approve when submitting the review:
selection_206

**Comment by LoisRForgeFlow** [comment]
@damendieta If you are ok with this we can merge it I think

**Review by SimoRubi — APPROVED** [review]
Code review, looks good

**Review by andhit-r — APPROVED** [review]

**Pull request merged**
